### PR TITLE
Finite range directional global potentials

### PR DESCRIPTION
**Pull request by jws-1**
This PR adds a new category of global potentials: Directional Global Potentials. Moreover, it adds a Directional Global Potential form, `LJCylinder` - which produces a cylindrical Lennard-Jones Potential defined by an origin point and a vector describing the "direction" of the cylinder. Directional Global Potentials are interfaced with through a new node of the same name.

### CODE REVIEW

**Review by trisyoungs — COMMENTED** [review]
Added a commit of my own - apologies!  I had to make the changes to do the testing and it was just easier to commit the changes direct.

All looking good here - one comment on the math to consider.

File: src/kernels/potentials/directional.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    // Compute new components
+    v.x = vector_.y * xyyx + vector_.z * xzzx;
+    v.y = vector_.z * yzzy - vector_.x * xyyx;
+    v.z = -vector_.x * xzzx - vector_.y * yzzy;
+
+    auto vecji = box->minimumImage(v, Vec3<double>{0., 0., 0.});
+
+    // Minimum distance between the atom and a point on the line
+    auto r = vecji.magnitude();
```
[the remark this answers is not in the document]
Comment (jws-1) [reply src/kernels/potentials/directional.cpp:97]: To be honest, I wasn't entirely sure what it was for either... I assumed that it ensured the position was w.r.t the origin of the box - but I guess that's not necessary here so I'll remove it.

**Comment by jws-1** [comment]
> Added a commit of my own - apologies! I had to make the changes to do the testing and it was just easier to commit the changes direct.
> 
> All looking good here - one comment on the math to consider.

Thanks for doing the work for me! :laughing:

File: src/kernels/potentials/directional.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+     * Definition
+     */
+
+    private:
+    // Potential form
+    InteractionPotential<DirectionalPotentialFunctions> interactionPotential_;
+    // Coordinate origin of potential
+    Vec3<double> origin_;
+    // Direction of potential
+    Vec3<double> vector_;
```
Comment (trisyoungs) [inline src/kernels/potentials/directional.h:44]: Just adding a sensible default.
```suggestion
    Vec3<double> vector_{0.0,0.0,1.0};
```

**Review by trisyoungs — APPROVED** [review]
:+1:

**Pull request merged**